client_with_class.py

```diff
@@ -0,0 +1,134 @@
+import keyboard
+import socket
+import time
+from abc import ABC, abstractmethod
+from pynput import mouse
+from threading import Thread
+
+
+class MetricCollector(ABC):
+    def __init__(self, name):
+        self.name = name
+        self.value = 0
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._running = False
+
+    def start_collect(self, host, port, interval=60):
+        self.interval = interval
```

Comment:
Please check this point: `Instance attribute interval defined outside __init__` for some properties of Class

Reply:
Fixed